### PR TITLE
feat: add support for ESLint v8

### PR DESCRIPTION
**Pull request by marcusrbrown**
## Description

Adds minimal support for ESLint v8. A few dependencies required updates to versions that are compatible with ESLint v8.

### Breaking Changes?

A project that uses ESLint v7 with `eslint-config-amex` should not be affected by this change. 

The [breaking change](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/CHANGELOG.md#1900--2021-11-10) in `eslint-config-airbnb` v19 removes support for ESLint < v7. `eslint-config-amex` requires a minimum of ESLint v7.

The `eslint-plugin-jest` upgrade to v25 introduces [breaking changes](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.0.0) to `jest:recommended` and updates rules that could be in use by projects that use the `test` config preset. I widened the peer dependency's version range to leave v24 intact. Users who arrive here on ESLint v8 must use `eslint-plugin-jest@^25.1.0` or later.

`eslint-plugin-unicorn` [v37 release](https://github.com/sindresorhus/eslint-plugin-unicorn/releases/tag/v37.0.0) does not introduce breaking changes when updating from v36.

Please make sure that the PR fulfills these requirements

- [x] I checked that there aren't other open Pull Requests for the same update/change.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] Rule changes includes a comment to describe the reasoning behind the change.
- [ ] PR contains a single rule change.
    
## Motivation and Context

The current major version of ESLint must be supported by all config presets. We should continue to support users on v7. I would also recommend updating the unit test to accept any ESLint version in dev deps that satisfies the peer dep range. I will followup with that change later.

## How Has This Been Tested?

`npm test`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [x] Dependency update

### CODE REVIEW

**Review by code-forger — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -43,21 +43,21 @@
...
     "eslint-plugin-react": "^7.21.5",
     "eslint-plugin-react-hooks": "^4.2.0",
-    "eslint-plugin-unicorn": "^36.0.0",
+    "eslint-plugin-unicorn": "^37.0.1",
     "eslint-plugin-you-dont-need-lodash-underscore": "^6.10.0"
   },
   "peerDependencies": {
-    "eslint": "^7.32.0",
-    "eslint-plugin-jest": "^24.1.3",
+    "eslint": "^7.32.0 || ^8.8.0",
```
Comment (Matthew-Mallimo) [inline package.json:59]: Will `8.x` work as well, or do we specifically need `^8.8.0` ?

Comment (marcusrbrown) [reply package.json:59]: 8.8.0 was out when I created the PR, what is the benefit of widening it?

Comment (Matthew-Mallimo) [reply package.json:59]: It forces packages to have either v7.32.0 or 8.8.0. If 8.8.1 comes out this repo would need another PR

**Review by 10xLaCroixDrinker — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (10xLaCroixDrinker) [reply package.json:59]: @Matthew-Mallimo with the `^`, that is not an issue

**Comment by 10xLaCroixDrinker** [comment]
@marcusrbrown can you resolve conflicts?

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Pull request merged**